digitaltwins/model_repository/src/cl/model_repository_service_client.ts

```diff
@@ -0,0 +1,280 @@
+          xMsRequestId: response ? response.parsedHeaders ? response.parsedHeaders.xMsRequestId : undefined : undefined,
+          eTag: response ? response.parsedHeaders ? response.parsedHeaders.eTag : undefined : undefined,
+        };
+      _callback(err as Error, createOrUpdateModelResponse, response);
+      });
+    }, (createOrUpdateModelResponse, response) => createOrUpdateModelResponse, callback as TripleValueCallback<CreateOrUpdateModelResponse, msRest.HttpOperationResponse>);
+  }
+
+  /**
+   * @method deleteModel                             module: azure-iot-modelrepository-service.ModelRepoServiceClient.deleteModel
+   * @description                                    Deletes a Digital Twin Model in the Model Repository.
+   * @param {string} modelId                         The model Id of the model to delete.
+   * @returns DeleteModelResponse                    The return object containing the DeleteResponse plus the HttpResponse.
+   * @memberof ModelRepositoryServiceClient
+   */
+  deleteModel(modelId: string, options?: DeleteModelParams): Promise<DeleteModelResponse>;
+  deleteModel(modelId: string, options?: DeleteModelParams, callback?: TripleValueCallback<DeleteModelResponse, msRest.HttpOperationResponse>): void;
+  deleteModel(modelId: string, options?: DeleteModelParams, callback?: TripleValueCallback<DeleteModelResponse, msRest.HttpOperationResponse>): void | Promise<DeleteModelResponse> {
+    /* Codes_SRS_NODE_MODEL_REPOSITORY_SERVICE_CLIENT_12_018: [ The `deleteModel` method shall call the `deleteModel` method of the protocol layer with the given arguments. ]*/
+    /* Codes_SRS_NODE_MODEL_REPOSITORY_SERVICE_CLIENT_12_019: [ The `deleteModel` method shall call the callback with an error parameter if a callback is passed. ]*/
+    /* Codes_SRS_NODE_MODEL_REPOSITORY_SERVICE_CLIENT_12_020: [ The `deleteModel` method shall return error if the method of the protocol layer failed. ]*/
+    /* Codes_SRS_NODE_MODEL_REPOSITORY_SERVICE_CLIENT_12_021: [ The `deleteModel` method shall return a promise if there is no callback passed. ]*/
+    return tripleValueCallbackToPromise<DeleteModelResponse, msRest.HttpOperationResponse, DeleteModelResponse>((_callback) => {
+      if (!options) {
+          options = {
+            'x-ms-client-source': this._clientSource,
+            'xMsClientRequestId': uuid.v4()
+        };
+      }
+      this._pl.deleteModel(modelId, this._credentials.getRepositoryId(), this._apiVersion, options, (err, result, request, response) => {
+        let deleteModelResponse: DeleteModelResponse = {
+          _response: response ? response : undefined,
+          xMsRequestId: response ? response.parsedHeaders ? response.parsedHeaders.xMsRequestId : undefined : undefined
+        };
+      _callback(err as Error, deleteModelResponse, response);
+      });
+    }, (deleteModelResponse, response) => deleteModelResponse, callback as TripleValueCallback<DeleteModelResponse, msRest.HttpOperationResponse>);
```

Comment:
why do we pass back deleteModelResponse AND response, when deleteModelResponse contains response? Seems redundant. Can this be changed?